### PR TITLE
feat(3d-viewer): added new component

### LINKED ISSUE #1674: 3d-viewer: new component
Add new component for 3d viewer

### PR DESCRIPTION
**Pull request by agliga**
Closes #1674

<br class="Apple-interchange-newline">

## Description
Added new 3d viewer component

## Context
* Changed the loader code to be put in common and have video and 3d viewer rely on it
* 3d viewer loads both source and a webworker placeholder to execute the code (It sepcifically relys on the ID of the webworker script element to load)
* It uses assetID which is stored on the restar site to render the 3d view

## Screenshots
Screen Shot 2022-03-22 at 1 43 01 PM
Screen Shot 2022-03-22 at 1 43 07 PM
Screen Shot 2022-03-22 at 1 43 35 PM
Screen Shot 2022-03-22 at 1 43 42 PM

### CODE REVIEW

**Comment by ianmcburnie** [comment]
It's great that keyboard is working out of the box. Either ARROW keys or Q-W-E-A-S-D seem to be working well. For arrow keys I see that the event still bubbles up and scrolls the main page - we might need to prevent propagation of those keys if we can (similar to what [makeup-prevent-scroll-keys](https://github.com/makeup/makeup-js/tree/master/packages/makeup-prevent-scroll-keys)) does.

**Review by ArtBlue — APPROVED** [review]
Looks great!

File: src/components/ebay-3d-viewer/component.js [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+                dampingFactor = 0.12,
+            } = this.input;
+
+            const allowViewBottom = !this.input.noAllowViewButtom;
+            const autoRotate = !this.input.noAutoRotate;
+            const mouseMoveFollow = !this.input.noMouseMoveFollow;
+            const animatedPlacement = !this.input.noAnimatedPlacement;
+
+            // eslint-disable-next-line no-undef,new-cap
+            this.rest = RestARViewer({
```
[the remark this answers is not in the document]
Comment (agliga) [reply src/components/ebay-3d-viewer/component.js:79]: I don't have access or know where their documentation is. 
I will check and add it later when i add other doc changes.

**Review by ianmcburnie — APPROVED** [review]
Groovy!

**Pull request merged**